pybossa/api/__init__.py

```diff
@@ -417,6 +420,82 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name):
+    """API endpoint for loading account user preferences.
+    Returns a JSON object containing the user account preferences.
+    """
+    user = user_repo.get_by_name(user_name)
+    if not user:
+        return abort(404)
+
+    try:
+        (can_update, disabled_fields, hidden_fields) = can_update_user_info(current_user, user)
```

Comment:
Users without (sub)admin privilege can read their user preference from web API `api/user/id`. Should we keep our rules consistent?

Reply:
Just to confirm, is the proposal to allow a user to read their task preferences regardless of the value of `disabled_fields` or `hidden_fields`. But, only allow admins/subadmins to update the task preferences?